fixups/FileRedirectionFixup/PathRedirection.cpp

```diff
@@ -507,16 +579,24 @@ std::string UrlDecode(std::string str)
             ret += str[i];
         }
         else 
-        {
-            sscanf_s(str.substr(i + 1, 2).c_str(), "%x", &ii);
-            ch = static_cast<char>(ii);
-            ret += ch;
-            i = i + 2;
+        {            
+            if (sscanf_s(str.substr(i + 1, 4).c_str(), "%x", &ii) == 1)

```

Comment:
What is the following loop supposed to accomplish.  I can't figure it out.

Reply:
The purpose of the method is to replacing references like "%20" in a string.  This is done in a loop as there may be more than one instance in the input string. You may have a more standard function to do this somewhere, but I didn't find one.

Reply:
I noticed that there are two places where this logic is used.

For now can you please add a comment explaining what this does?  If it turns out we need to add this logic in again somewhere let's move it into a method.

Reply:
Done.